### DISCUSSION TITLE
求助！向量查询出是无关数据

### DISCUSSION THREAD

**Discussion by lsy88** [milvus-io/milvus#38437] [Q&A and General discussion]
我目前将一些md文章通过切割后嵌入milvus中，但是我在查询时，返回的都是一些无关数据，有人能给我一些提示或帮助吗

**Comment by yhmo** [comment]
跟模型的关系比较大。入库时所使用的模型和查询时使用的模型相同，那么milvus就会严格根据既定算法计算相似度。

**Reply by lsy88** [reply]
我在入库和查询时使用的嵌入模型都是all-MiniLM-L12-v2这个

**Reply by yhmo** [reply]
简单来说就是milvus只负责计算向量的相似度，你可以把它看成一个向量的计算器，输入什么就计算什么。如果输入的向量之间的距离它计算出来是正确的，那么它就没问题，不管这两条向量如何生成的，每条向量后面代表的语义是什么它是不管的。
无关数据被搜出来，要么模型不太适合，要么插入数据的时候错位了之类的。
